lib/maths/time_series/unittest/CSignalTest.cc

```diff
@@ -567,7 +567,7 @@ BOOST_AUTO_TEST_CASE(testReweightOutliers) {
 
     test::CRandomNumbers rng;
 
-    maths::time_series::CSignal::TFloatMeanAccumulatorVec values;
+    maths::time_series::CSignal::TFloatMeanAccumulatorVec values(144);
```

Comment:
@tveasey of the 4 tests that were warning of no assertions this is the one that's a worry.  This test genuinely was not doing what it should have been doing, because everything was based on `values.size()` and `values.size()` was 0.  I copied the 144 here from a different test that was using the same pattern of basing everything on `values.size()`, but now the test fails. 🙀 

The failure is:

```
/Users/dave/ml-cpp/lib/maths/time_series/unittest/CSignalTest.cc:596: fatal error: in "CSignalTest/testReweightOutliers": critical check maths::common::CBasicStatistics::count(values[i]) < 1.0 has failed [1 >= 1]
```

Since this test seems to have been doing nothing up to now, please could you have a look at what it should be asserting now.

Reply:
Whoops! The issue is you can't use 144 for values size, otherwise 30 isn't an outlier for the second component. I would leave this line unchanged and update the line where we assign to values to be:
```
values.assign(100, maths::time_series::CSignal::TFloatMeanAccumulator{});
```